core/extension/gdextension_library_loader.cpp

```diff
@@ -195,16 +195,21 @@ Error GDExtensionLibraryLoader::open_library(const String &p_path) {
 		&abs_dependencies_paths, // library_dependencies
 	};
 
-	err = OS::get_singleton()->open_dynamic_library(is_static_library ? String() : abs_path, library, &data);
+	// Apple has a complex lookup system which goes beyond looking up the filename, so we try that first
```

Comment:
Nitpick, not a blocker:
```suggestion
	// Apple has a complex lookup system which goes beyond looking up the filename, so we try that first.
```